packages/app-page-builder/src/types.ts

```diff
@@ -208,8 +208,9 @@ export interface PbElement {
 
 export interface PbBlockVariable {
     id: string;
+    type: string;
     label: string;
-    value?: string;
+    value?: any;
 }
```

Comment:
It's probably a good idea to make this interface generic, so you can use the default interface where the variable type is unknown, but use a concrete data type within the rendering logic:

```ts
export interface PbBlockVariable<TValue = any> {
    id: string;
    type: string;
    label: string;
    value?: TValue;
}
```

Then you can use it like this:

```ts
type ButtonVariable = {
  label: string;
  url: string;
}

const variable: PbBlockVariable<ButtonVariable> = element.whatever;

// variable.value.label & variable.value.url are now properly typed and provide autocomplete.
```